### PR TITLE
fs/epoll: change type of eventset from uint8_t to uint32_t

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
fs/epoll: change type of eventset from uint8_t to uint32_t to support EPOLLONESHOT and so on.

Signed-off-by: Jiuzhu Dong <dongjiuzhu1@xiaomi.com>
## Impact
support EPOLLHUP,EPOLLRDHUP,EPOLLONESHOT...
## Testing
local test

### CODE REVIEW

File: fs/vfs/Kconfig [hunk 1]
```diff
@@ -60,3 +60,9 @@ config TIMER_FD_NPOLLWAITERS
 		Maximum number of threads that can be waiting on poll()
 
 endif # TIMER_FD
+
+config EPOLL_FULLEVENTS
```
Comment (xiaoxiang781216) [inline fs/vfs/Kconfig:64]: let's change events type instead add config option

Comment (Donny9) [reply fs/vfs/Kconfig]: Okay, Done

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/cxd56xx/cxd56_geofence.c [hunk 2]
```diff
@@ -450,7 +450,7 @@ static void cxd56_geofence_sighandler(uint32_t data, FAR void *userdata)
       if (fds)
         {
           fds->revents |= POLLIN;
-          gnssinfo("Report events: %02x\n", fds->revents);
+          gnssinfo("Report events: %02" PRIx32 "\n", fds->revents);
```
Comment (pkarashchenko) [inline arch/arm/src/cxd56xx/cxd56_geofence.c:453]:
```suggestion
          gnssinfo("Report events: %08" PRIx32 "\n", fds->revents);
```
here and in other places

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@Donny9 please fix the printf warning.

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 mainline CI fails with
```
In file included from sensors/sensor.c:33:
sensors/sensor.c: In function 'sensor_pollnotify':
Error: sensors/sensor.c:170:22: error: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'pollevent_t' {aka 'long unsigned int'} [-Werror=format=]
  170 |               sninfo("Report events: %02x\n", fd->revents);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~
      |                                                 |
      |                                                 pollevent_t {aka long unsigned int}
sensors/sensor.c:170:41: note: format string is defined here
  170 |               sninfo("Report events: %02x\n", fd->revents);
      |                                      ~~~^
      |                                         |
      |                                         unsigned int
      |                                      %02lx
cc1: all warnings being treated as errors
make[1]: *** [Makefile:91: sensor.o] Error 1
```

**Comment by yamt** [comment]
> fs/epoll: change type of eventset from uint8_t to uint32_t to support EPOLLONESHOT and so on.

can you explain a bit?
does linux support these EPOLL flags for poll(2)?

**Comment by xiaoxiang781216** [comment]
> > fs/epoll: change type of eventset from uint8_t to uint32_t to support EPOLLONESHOT and so on.
> 
> can you explain a bit? does linux support these EPOLL flags for poll(2)?

https://man7.org/linux/man-pages/man2/epoll_ctl.2.html
```
EPOLLONESHOT (since Linux 2.6.2)
              Requests one-shot notification for the associated file
              descriptor.  This means that after an event notified for
              the file descriptor by [epoll_wait(2)](https://man7.org/linux/man-pages/man2/epoll_wait.2.html), the file descriptor
              is disabled in the interest list and no other events will
              be reported by the epoll interface.  The user must call
              epoll_ctl() with EPOLL_CTL_MOD to rearm the file
              descriptor with a new event mask.

              This flag is an input flag for the event.events field when
              calling epoll_ctl(); it is never returned by
              [epoll_wait(2)](https://man7.org/linux/man-pages/man2/epoll_wait.2.html).
```

**Pull request merged**